nevergrad/functions/photonics/core.py

```diff
@@ -124,7 +125,8 @@ def __init__(self, name: str, dimension: int, transform: str = "tanh") -> None:
         self._func = inst.CommandFunction(["octave-cli", "--no-gui", "--no-history", "--norc", "--quiet", "--no-window-system", path.name],
                                           cwd=path.parent, verbose=False,
                                           env=dict(os.environ, OMP_NUM_THREADS="1", OPENBLAS_NUM_THREADS="1"))
-        super().__init__(self._compute, *_make_instrumentation(name=name, dimension=dimension, transform=transform).args)
+        super().__init__(self._compute, _make_instrumentation(name=name, dimension=dimension, transform=transform))
+        self.register_initialization(name=name, dimension=dimension, transform=transform)
```

Comment:
just to make it clear once: `register_initialization` keeps a record of the initialization parameters so that the "copy" method can reinstantiate a new object easily